docs/features/logforwarding/parse-json-structured-logs.adoc

```diff
@@ -0,0 +1,152 @@
+
+== Forwarding JSON Structured Logs
+
+Allows structured JSON log entries to be forwarded as JSON objects.
+
+=== ClusterLogForwarder configuration
+
+*pipelines* field:
+
+* `parse`: (string, optional) Currently the only allowed value is "json".
+If set, attempt to parse log entries as JSON objects.
+
+With the following forwarder configuration:
+
+.cluster-log-forwarder.yaml
+[source,yaml]
+----
+kind: ClusterLogForwarder
+apiVersion: logging.openshift.io/v1
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  pipelines:
+    - name: my-json-logs
+      outputRefs:
+        - default
+      inputRefs:
+        - application
+        - infrastructure
+      parse: json
+----
+
+* For each pipeline, if `parse: json` is set _and_ the log is valid JSON, the output record will include a `structured` field _equivalent_ to the JSON entry.
+
+
+=== Relationship between `structured` and `message` fields:
+
+* The original `message` field will be removed when the new `structured` field is included.
+* If there is no valid JSON data found, the `structured` field will not be present.
+* Both `message` and `structured` fields _should not_ be present in the same entry.
+
+For example, an application sends this structured log message:
+
+[source,json]
+----
+{"level":"info","method":"get","source":"homepage"}
+----
+
+Typically, the forwarded log record looks like this:
+
+[source,json]
+----
+{"message":"{\"level\":\"info\",\"method\":\"get\",\"source\":\"homepage\"}",
+ "more fields..."}
+----
+
+The new record with JSON parsing enabled replaces `message` field with a `structured` object:
+
+[source,json]
+----
+{"structured": {"level":"info","method":"get","source":"homepage"},
+ "more fields..."}
+----
+
+
+
+
+=== Output type: Elasticsearch
+
+With Elasticsearch, JSON records with different formats *must* go to different indices, otherwise type conflicts and cardinality problems can occur.
```

Comment:
We should probably clarify 'different formats'.  The gist is like a database schema.  Maybe we can say something like same sets of keys for values of the same type???